go/arrow/flight/flightsql/example/sqlite_server.go

```diff
@@ -618,21 +618,21 @@ func getParamsForStatement(rdr flight.MessageReader) (params [][]interface{}, er
 	return params, rdr.Err()
 }
 
-func (s *SQLiteFlightSQLServer) DoPutPreparedStatementQuery(_ context.Context, cmd flightsql.PreparedStatementQuery, rdr flight.MessageReader, _ flight.MetadataWriter) error {
+func (s *SQLiteFlightSQLServer) DoPutPreparedStatementQuery(_ context.Context, cmd flightsql.PreparedStatementQuery, rdr flight.MessageReader, _ flight.MetadataWriter) ([]byte, error) {
 	val, ok := s.prepared.Load(string(cmd.GetPreparedStatementHandle()))
 	if !ok {
-		return status.Error(codes.InvalidArgument, "prepared statement not found")
+		return nil, status.Error(codes.InvalidArgument, "prepared statement not found")
 	}
 
 	stmt := val.(Statement)
 	args, err := getParamsForStatement(rdr)
 	if err != nil {
-		return status.Errorf(codes.Internal, "error gathering parameters for prepared statement query: %s", err.Error())
+		return nil, status.Errorf(codes.Internal, "error gathering parameters for prepared statement query: %s", err.Error())
 	}
 
 	stmt.params = args
 	s.prepared.Store(string(cmd.GetPreparedStatementHandle()), stmt)
```

Comment:
Not a Go expert, but is this caching the prepared statement with parameters against the original handle?
Should a stateless implementation provide a new handle, presumably containing query and parameters?

Reply:
Thank you for clarifying @zeroshade . Looks like I have a grasp of Go then.
I think if possible it would be worth having an example of stateless use in this PR. That might be a bigger change since this server implementation is based on stateful behaviour.
I guess I'm curious how we would create a new handle based on query and parameters or if we need a new proto message from which we can generate a handle.

Reply:
Because the handles are completely opaque to the protocol, it would depend entirely on how a given server chooses to represent its handle. The example server just uses a random character string as the handle which is a key into the map. But the handle could just as easily be a serialized protobuf message, etc. if desired.

Reply:
Yeah, I get that it's opaque so really doesn't matter here. I'll post a Java implementation PR soon with a stateless example which might be useful, but very much down to individuals what they might pick from it.

Reply:
Yes. I decided an actual stateless implementation was out of scope for this change, and just made the minimal change to achieve compatibility with the new spec, which allows for either stateless or stateful server implementation. There is certainly an argument to be made for serverless implementation, but I leave that decision up to the maintainers.

Reply:
The title of this PR is misworded then, so I will update it to reflect what the change is.